src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2658,19 +2658,26 @@ private void EnsureTypeParametersAreLoaded()
             {
                 if (_lazyTypeParameters.IsDefault)
                 {
+                    if (_arity == 0)
+                    {
+                        _lazyTypeParameters = ImmutableArray<TypeParameterSymbol>.Empty;
```

Comment:
>_lazyTypeParameters = ImmutableArray<TypeParameterSymbol>.Empty; [](http://example.com/codeflow?start=24&length=64)

Strictly speaking, this assignment is not guaranteed to be atomic. Please use ```ImmutableInterlocked.InterlockedInitialize```.

Reply:
As an alternative, we could check arity and assign this way inside the constructor. There is a guarantee that there is no race with any other thread,

Reply:
I would prefer that approach myself.

Reply:
> I would prefer that approach myself.

I assume "that" refers to the second comment? https://github.com/dotnet/roslyn/pull/70766/commits/2456a067124af5b4fce55bf813b11922b4901bb5